packages/notifi-react/lib/context/NotifiTargetContext.tsx

```diff
@@ -246,25 +276,44 @@ export const NotifiTargetContextProvider: FC<PropsWithChildren> = ({
     },
     [],
   );
-  console.log('targetGroupToBeSaved', targetGroupToBeSaved);
-  const renewTargetGroup = useCallback(() => {
-    setIsLoading(true);
-    console.log('inside', targetGroupToBeSaved);
-    return frontendClient
-      .ensureTargetGroup(targetGroupToBeSaved)
-      .then((_result) => {
-        frontendClient
-          .fetchData()
-          .then((data) => {
-            refreshTargetDocument(data);
-            setError(null);
-          })
-          .catch((e) => setError(e as Error))
-          .finally(() => setIsLoading(false));
-      })
-      .catch((e) => setError(e as Error))
-      .finally(() => setIsLoading(false));
-  }, [frontendClient, targetGroupToBeSaved]);
+
+  const renewTargetGroup = useCallback(
+    (data?: TargetGroupInput) => {
+      setIsLoading(true);
+      data = data ?? targetGroupToBeSaved;
+      return frontendClient
+        .ensureTargetGroup(data)
+        .then((_result) => {
+          frontendClient
+            .fetchData()
+            .then((data) => {
+              refreshTargetDocument(data);
+              setError(null);
+            })
+            .catch((e) => setError(e as Error))
+            .finally(() => setIsLoading(false));
+        })
+        .catch((e) => setError(e as Error))
+        .finally(() => setIsLoading(false));
+    },
+    [frontendClient, targetGroupToBeSaved],
+  );
+
+  const renewToggleTargetGroup = useCallback(
+    async (
+      target: Extract<Target, 'discord' | 'slack' | 'wallet'>,
+      value: boolean,
```

Comment:
Rather than creating a new `renewToggleTargetGroup`, do you think it is possible to combine this one to existing `renewTargetGroup`? I am afraid exposing two interfaces might cause confusion.

As far as I concern, all the updated un-updated data is already in `targetGroupToBeSaved`, I would rather just add one more optional argument in `renewTargetGroup` to only renew single target if that is the case.